### PR TITLE
rpc: Introduce `verifyislock`

### PR DESCRIPTION
**Pull request by UdjinM6**
~Based on #3912 atm~

### CODE REVIEW

File: test/functional/rpc_verifyislock.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        node.spork("SPORK_17_QUORUM_DKG_ENABLED", 0)
+        self.wait_for_sporks_same()
+
+        self.mine_quorum()
+
+        txid = node.sendtoaddress(node.getnewaddress(), 1)
+        self.wait_for_instantlock(txid, node)
+
+        tx = FromHex(CTransaction(), node.getrawtransaction(txid))
+        tx.rehash()
```
Comment (xdustinface) [inline test/functional/rpc_verifyislock.py:35]:
```suggestion
```

Could be dropped, hash is not really needed here.

**Review by xdustinface — CHANGES_REQUESTED** [review]
I guess referenced commit in description was meant to be 592238faf9e87d94ad62a40d88973d55f08295bc instead of 91905ce? I would say this should be either mentioned in the title, moved to a seperate PR or we merge this PR without squashing?

Also see one suggestion below

File: test/functional/rpc_verifyislock.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+        request_id_buf = ser_string(b"islock") + ser_compact_size(len(tx.vin))
+        inputs = []
```
Comment (xdustinface) [inline test/functional/rpc_verifyislock.py:38]:
```suggestion
```

File: test/functional/rpc_verifyislock.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        for txin in tx.vin:
+            request_id_buf += txin.prevout.serialize()
+            inputs.append(txin.prevout)
```
Comment (xdustinface) [inline test/functional/rpc_verifyislock.py:41]:
```suggestion
```

**Review by xdustinface — CHANGES_REQUESTED** [review]
The two below can also be dropped

**Comment by UdjinM6** [comment]
Rebased, `auto->int` for `signOffset`, squashed.

**Review by xdustinface — APPROVED** [review]
utACK

File: test/functional/rpc_verifyislock.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env python3
+# Copyright (c) 2015-2020 The Dash Core developers
```
Comment (PastaPastaPasta) [inline test/functional/rpc_verifyislock.py:2]:
```suggestion
# Copyright (c) 2020 The Dash Core developers
```

**Review by PastaPastaPasta — CHANGES_REQUESTED** [review]

**Review by xdustinface — APPROVED** [review]
utACK. I honestly don't have a very strong opinion about those copyright years so im good with 2020 (because it was implemented there?).. but shouldn't we use 2021 or at least "2020-2021" now to be accurate? :D

**Review by PastaPastaPasta — APPROVED** [review]
utACK

**Pull request merged**